### PR TITLE
Activity log is now an object, with context

### PR DESCRIPTION
**Pull request by rtxm**
We can transmit it down the component tree, adding context by the way, so that any component can log activity without refering to upper levels.
This is part of #98.

### CODE REVIEW

File: kansha/board/models.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 from kansha.column.models import DataColumn
```
Comment (vrialland) [inline kansha/board/models.py]: Import order

File: kansha/column/comp.py [hunk 2]
```diff
@@ -250,16 +256,16 @@ def delete_card(self, c):
         """
         self.cards = [card for card in self.cards if c != card()]
         values = {'column_id': self.id, 'column': self.get_title(), 'card': c.get_title()}
-        notifications.add_history(self.board.data, c.data,
-                                  security.get_user().data,
-                                  u'card_delete', values)
+        c.action_log.add_history(
+            security.get_user(),
+            u'card_delete', values)
         self.search_engine.delete_document(fts_schema.Card, c.id)
```
Comment (vrialland) [inline kansha/column/comp.py]: "c" is not an explicit var name

File: kansha/column/comp.py [hunk 3]
```diff
@@ -250,16 +256,16 @@ def delete_card(self, c):
...
         self.search_engine.commit()
         c.delete()
 
     def refresh(self):
         self.cards = [component.Component(
-            self._services(card.Card, c.id, self, self.card_extensions, data=c)
+            self._services(card.Card, c.id, self, self.card_extensions, self.action_log, data=c)
             ) for c in self.data.cards]
```
Comment (vrialland) [inline kansha/column/comp.py]: "c" is not an explicit var name

File: kansha/column/comp.py [hunk 4]
```diff
@@ -270,7 +276,7 @@ def archive_card(self, c):
         """
         self.cards = [card for card in self.cards if c != card()]
         values = {'column_id': self.id, 'column': self.get_title(), 'card': c.get_title()}
-        notifications.add_history(self.board.data, c.data, security.get_user().data, u'card_archive', values)
+        c.action_log.add_history(security.get_user(), u'card_archive', values)
         self.board.archive_card(c)
```
Comment (vrialland) [inline kansha/column/comp.py]: "c" is not an explicit var name

File: kansha/column/comp.py [hunk 5]
```diff
@@ -270,7 +276,7 @@ def archive_card(self, c):
...
 
     def edit_card(self, c):
```
Comment (vrialland) [inline kansha/column/comp.py]: "c" is not an explicit var name

File: kansha/services/actionlog/models.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+# All rights reserved.
+#
+# This software is licensed under the BSD License, as described in
+# the file LICENSE.txt, which you should have received as part of
+# this distribution.
+# --
+
+from datetime import datetime, timedelta
+import json
+
```
Comment (vrialland) [inline kansha/services/actionlog/models.py]: Import order

File: kansha/services/actionlog/models.py [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    action = Field(Unicode(255))
+    data = Field(JSONType)
+
+    board = ManyToOne('DataBoard', ondelete='cascade')
+    card = ManyToOne('DataCard', ondelete='cascade')
+    user = ManyToOne('DataUser', ondelete='cascade')
+
+    def to_string(self):
+        print 'ouh!'
+        data = self.data.copy()
```
Comment (vrialland) [inline kansha/services/actionlog/models.py]: print statement

File: kansha/column/comp.py (hunk 2 above)
Comment (rtxm) [inline kansha/column/comp.py]: That function is completely rewritten in another PR.

File: kansha/column/comp.py (hunk 4 above)
Comment (rtxm) [inline kansha/column/comp.py]: Function entirely rewritten in another PR.

File: kansha/column/comp.py (hunk 5 above)
Comment (rtxm) [inline kansha/column/comp.py]: `edit_card` disappears in another PR.

**Pull request merged**